test_conformance/extensions/cl_khr_semaphore/test_semaphores_negative_create.cpp

```diff
@@ -0,0 +1,672 @@
+    {}
+
+    cl_int Run() override
+    {
+        if (!is_extension_available(device,
+                                    "cl_khr_external_semaphore_opaque_fd"))
+        {
+            log_info(
+                "cl_khr_external_semaphore_opaque_fd is not supported on this "
+                "platoform. Skipping test.\n");
+            return TEST_SKIPPED_ITSELF;
+        }
+
+        cl_int err = CL_SUCCESS;
+        clCommandQueueWrapper queue = clCreateCommandQueue(
+            context, device, CL_QUEUE_OUT_OF_ORDER_EXEC_MODE_ENABLE, &err);
+        test_error(err, "Could not create command queue");
+
+        // Create semaphore
+        cl_semaphore_properties_khr sema_1_props[] = {
+            (cl_semaphore_properties_khr)CL_SEMAPHORE_TYPE_KHR,
+            (cl_semaphore_properties_khr)CL_SEMAPHORE_TYPE_BINARY_KHR,
+            (cl_semaphore_properties_khr)CL_SEMAPHORE_EXPORT_HANDLE_TYPES_KHR,
+            (cl_semaphore_properties_khr)CL_SEMAPHORE_HANDLE_OPAQUE_FD_KHR,
+            (cl_semaphore_properties_khr)
+                CL_SEMAPHORE_EXPORT_HANDLE_TYPES_LIST_END_KHR,
+            0
+        };
+        semaphore =
+            clCreateSemaphoreWithPropertiesKHR(context, sema_1_props, &err);
+        test_error(err, "Could not create semaphore");
+
+        // Signal semaphore
+        clEventWrapper signal_event;
+        err = clEnqueueSignalSemaphoresKHR(queue, 1, semaphore, nullptr, 0,
+                                           nullptr, &signal_event);
+        test_error(err, "Could not signal semaphore");
+
+        // Extract sync fd
+        int handle = -1;
+        size_t handle_size;
+        err = clGetSemaphoreHandleForTypeKHR(
+            semaphore, device, CL_SEMAPHORE_HANDLE_OPAQUE_FD_KHR,
+            sizeof(handle), &handle, &handle_size);
+        test_error(err, "Could not extract semaphore handle");
+        test_assert_error(sizeof(handle) == handle_size, "Invalid handle size");
+        test_assert_error(handle >= 0, "Invalid handle");
+
+        // Create semaphore from sync fd. Exporting a semaphore handle from
+        // a semaphore that was created by importing an external semaphore
+        // handle is not permitted.
+        cl_semaphore_properties_khr sema_2_props[] = {
+            (cl_semaphore_properties_khr)CL_SEMAPHORE_TYPE_KHR,
+            (cl_semaphore_properties_khr)CL_SEMAPHORE_TYPE_BINARY_KHR,
+            (cl_semaphore_properties_khr)CL_SEMAPHORE_HANDLE_OPAQUE_FD_KHR,
+            (cl_semaphore_properties_khr)(handle),
+            (cl_semaphore_properties_khr)CL_SEMAPHORE_EXPORT_HANDLE_TYPES_KHR,
+            (cl_semaphore_properties_khr)CL_SEMAPHORE_HANDLE_OPAQUE_FD_KHR,
+            (cl_semaphore_properties_khr)
+                CL_SEMAPHORE_EXPORT_HANDLE_TYPES_LIST_END_KHR,
+            0
+        };
+
+        semaphore_second =
+            clCreateSemaphoreWithPropertiesKHR(context, sema_2_props, &err);
+        test_failure_error(
+            err, CL_INVALID_DEVICE,
```

Comment:
Shouldn't this be **CL_INVALID_OPERATION**?

Reply:
Corrected, thanks!